test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/HistoryTests.cs

```diff
@@ -222,28 +258,56 @@ public async Task GivenAValueForSinceAndBeforeCloseToLastModifiedTime_WhenGettin
             var sinceUriString = HttpUtility.UrlEncode(since.ToString("o"));
 
             // Query all the recent changes
-            using FhirResponse<Bundle> allChanges = await _client.SearchAsync($"_history?_tag={tag}&_since={sinceUriString}&_count={count}");
+            FhirResponse<Bundle> response;
+            List<Bundle.EntryComponent> allChanges = new List<Bundle.EntryComponent>();
+            string searchString = $"_history?_since={sinceUriString}";
+            while (!string.IsNullOrEmpty(searchString))
+            {
+                response = await _client.SearchAsync(searchString);
+                allChanges.AddRange(response.Resource.Entry);
+                searchString = response.Resource.NextLink?.ToString();
+            }
```

Comment:
Nit: this block of code seems to be reused a lot - maybe we could move it to a common method?